src/viking.js

```diff
@@ -1,14 +1,160 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack(){
+    return this.strength;
+  }
+  receiveDamage(theDamage){
+    this.health = this.health - theDamage;
+  }
+}
+
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(theDamage){
+    this.health = this.health - theDamage;
+    if(this.health <= 0){
+      return `${this.name} has died in act of combat`;
+      
+    }
+    else {
+      return `${this.name} has received ${theDamage} points of damage`;
+    }
```

Comment:
nesses casos vc nem precisa usar "else" se quiser, depois do if se quiser mandar direto o "return ${this.name} has died in act of combat", também da certo 😉